src/common/console_service.cpp

```diff
@@ -120,91 +194,20 @@ ConsoleService::ConsoleService()
 
 ConsoleService::~ConsoleService()
 {
-    m_consoleThreadRun = false;
-
-    if (m_consoleInputThread.joinable())
-    {
-        m_consoleInputThread.join();
-    }
+    stop();
+    m_consoleStopCondition.notify_all();
 }
 
 // NOTE: If you capture things in this function, make sure they're protected (locked or atomic)!
 // NOTE: If you're going to print, use fmt::print, rather than ShowInfo etc.
 void ConsoleService::RegisterCommand(std::string const& name, std::string const& description, std::function<void(std::vector<std::string>)> func)
 {
-    std::lock_guard lock(m_consoleInputBottleneck);
+    std::lock_guard<std::mutex> lock(m_consoleInputBottleneck);
+
     m_commands[name] = ConsoleCommand{ name, description, func };
 }
 
 void ConsoleService::stop()
 {
     m_consoleThreadRun = false;
 }
-
-std::string ConsoleService::getLine()
-{
-    std::string line;
-
-// Windows doesn't have a proper poll that works for stdin, but we can use some win32 API on the console...
-#ifdef _WIN32
-    INPUT_RECORD record;
-    DWORD        numEvents;
-
-    while (m_consoleThreadRun)
-    {
-        if (!ReadConsoleInput(GetStdHandle(STD_INPUT_HANDLE), &record, 1, &numEvents))
-        {
-            continue; // this is an error state, does this happen in the real world?
-        }
-
-        if (record.EventType != KEY_EVENT)
-        {
-            continue;
-        }
-
-        if (record.Event.KeyEvent.bKeyDown) // only take input on keydown, not key up
-        {
-            WCHAR keyCharacter = record.Event.KeyEvent.uChar.UnicodeChar;
-
-            if (keyCharacter == '\b')
-            {
-                fmt::print("\b \b"); // move cursor left, overwrite with space, move cursor left
-                if (line.size() > 0)
-                {
-                    line.pop_back(); // remove last char in buffer if any
-                }
-                continue;
-            }
-
-            fmt::print("{:c}", keyCharacter); // echo character back to console, apparently using ReadConsoleInput & GetStdHandle prevents echo?
-            if (keyCharacter == '\r')
-            {
-                fmt::print("\n"); // Windows needs \r\n for newlines in the console, but the enter key is only \r.
-                break;
-            }
-
-            if (std::isprint(keyCharacter))
-            {
-                line += keyCharacter;
-            }
-        }
-    }
-// Linux has a proper polling system for stdin
-#else
-    struct pollfd pollFileDescriptor = { STDIN_FILENO, POLLIN, 0 };
```

Comment:
Maybe bringing back your impl with a 0 wait might have the same effect as what I'm using?

Reply:
I changed it to poll, but it had the same problem. I believe this is because windows returns stdin is ready for a character and linux returns stdin is ready on a newline via stdin file descriptos. I used std::getline on std::cin within an #IFDEF for non-windows and tested it, then added a commit.